virtualizarr/writers/kerchunk.py

```diff
@@ -87,8 +87,7 @@ def variable_to_kerchunk_arr_refs(var: Variable, var_name: str) -> KerchunkArrRe
             for chunk_key, entry in marr.manifest.dict().items()
         }
 
-        zarray = marr.zarray.replace(zarr_format=2)
-
+        array_v2_metadata = convert_v3_to_v2_metadata(marr.metadata)
```

Comment:
I don't think we want to go with this approach, perhaps we should allow for manifest arrays to have ArrayV2Metadata for writing to kerchunk.

Reply:
>  don't think we want to go with this approach

Why not? 

> perhaps we should allow for manifest arrays to have `ArrayV2Metadata` for writing to kerchunk.

I really don't see how that would make anything easier. Inside `variable_to_kerchunk_arr_refs` is the only place you should need to deal with v2 metadata, so why try to generalize it in the rest of the package if you know you won't need it anywhere else?

Reply:
that makes sense to me, I think I was just concerned about bugs trying to go between ArrayV3Metadata and ArrayV2Metadata. But having a single interface for metadata in memory is probably safer over all.

Reply:
Oh, I recalled, I am not sure yet about the best way to go from ArrayV3Codecs to V2 compressors and filters. I don't see anything yet in zarr-python for how to do that. I welcome any suggestions.

Reply:
Looking closer at https://github.com/zarr-developers/zarr-python/blob/main/src/zarr/core/array.py#L4290-L4334 it looks like we can parse the codecs in the ArrayV3Metadata:

```python
filters, serializer, codecs = array_v3_metadata.codecs
```